lib/scala/project-manager/src/test/scala/org/enso/projectmanager/protocol/ProjectManagementApiSpec.scala

```diff
@@ -313,6 +321,57 @@ class ProjectManagementApiSpec
       deleteProject(projectId)(client1)
     }
 
+    "start the Language Server after moving the directory" taggedAs Flaky in {
+      //given
+      val projectName     = "foo"
+      implicit val client = new WsTestClient(address)
+      val projectId       = createProject(projectName)
+
+      val newName       = "bar"
+      val newProjectDir = new File(userProjectDir, newName)
+      FileUtils.moveDirectory(
+        new File(userProjectDir, projectName),
+        newProjectDir
+      )
+      val packageFile = new File(newProjectDir, "package.yaml")
+      val mainEnso =
+        Paths.get(newProjectDir.toString, "src", "Main.enso").toFile
+      val meta =
+        Paths.get(newProjectDir.toString, ".enso", "project.json").toFile
+
+      packageFile shouldBe Symbol("file")
+      mainEnso shouldBe Symbol("file")
+      meta shouldBe Symbol("file")
+
+      //when
+      val socket = openProject(projectId)
+      val languageServerClient =
+        new WsTestClient(s"ws://${socket.host}:${socket.port}")
+      languageServerClient.send(json"""
+          {
+            "jsonrpc": "2.0",
+            "method": "file/read",
+            "id": 1,
+            "params": {
+              "path": {
+                "rootId": ${UUID.randomUUID()},
+                "segments": ["src", "Main.enso"]
+              }
+            }
+          }
+            """)
+      //then
+      languageServerClient.expectJson(json"""
+          {
+            "jsonrpc":"2.0",
+             "id":1,
+             "error":{"code":6001,"message":"Session not initialised"}}
+            """)
```

Comment:
It is unclear to me how an error indicates a successful test here.
I guess it's because if the project couldn't be opened the error would be different, but I think a short explanation would be really helpful for future maintainers.

Reply:
Thanks!